### PR TITLE
wlroots_0_18: init at 0.18.0

### PR DESCRIPTION
**Pull request by wineee**
## Description of changes

https://gitlab.freedesktop.org/wlroots/wlroots/-/releases/0.18.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/wlroots/default.nix [hunk 1]
```diff
@@ -137,5 +138,19 @@ rec {
...
+    ];
+    extraBuildInputs = [
+      ffmpeg
+      libliftoff
+      libdisplay-info
+      lcms2
+    ];
+  };
+
+  wlroots = wlroots_0_18;
```
Comment (adamcstephens) [inline pkgs/development/libraries/wlroots/default.nix:155]: Are you sure you want to change the default immediately?

Comment (wineee) [reply pkgs/development/libraries/wlroots/default.nix:155]: Yes, other packages should depend on `wlroots_0_*`

https://github.com/NixOS/nixpkgs/pull/245704#pullrequestreview-1553286765

**Comment by nim65s** [comment]
Result of `nixpkgs-review pr 330630` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>scenefx</li>
    <li>swayfx</li>
    <li>swayfx-unwrapped</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>nixpkgs-manual</li>
    <li>tinywl</li>
    <li>wlroots</li>
    <li>wlroots.examples</li>
  </ul>
</details>

**Review by nim65s — CHANGES_REQUESTED** [review]
I guess we need to fix wlroots version from swayfx & scenefx

**Comment by nim65s** [comment]
Also, only 0_14, 0_15 & 0_16 are exposed in `pkgs/top-level/all-packages.nix`. Is this on purpose ?

**Comment by nim65s** [comment]
Result of `nixpkgs-review pr 330630` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>tinywl</li>
    <li>wlroots (wlroots_0_18)</li>
    <li>wlroots.examples (wlroots_0_18.examples)</li>
  </ul>
</details>

**Review by nim65s — APPROVED** [review]

File: pkgs/development/libraries/wlroots/default.nix (hunk 1 above)
Comment (adamcstephens) [reply pkgs/development/libraries/wlroots/default.nix:155]: That makes me wonder why have an unversioned attribute at all then.

Comment (Synthetica9) [reply pkgs/development/libraries/wlroots/default.nix:155]: We typically used to follow Sway's wlroots version, and I think it still makes sense to expose wlroots-latest as an attribute, but we shouldn't use it internally. Maybe we can change it to an alias? Those are disalowed for use in nixpkgs afaik

**Review by adamcstephens — APPROVED** [review]

File: pkgs/development/libraries/wlroots/default.nix (hunk 1 above)
Comment (wineee) [reply pkgs/development/libraries/wlroots/default.nix:155]: Will change it to an alias in https://github.com/NixOS/nixpkgs/pull/252816

**Pull request merged**